src/network/networkd-ndisc.c

```diff
@@ -192,12 +260,62 @@ static int ndisc_router_process_default(Link *link, sd_ndisc_router *rt) {
         return 0;
 }
 
+static int ndisc_router_generate_address(Link *link, unsigned prefixlen, uint32_t lifetime_preferred, Address *address) {
+        bool prefix_stable = false;
+        struct in6_addr addr;
+        IPv6Token *j;
+        Iterator i;
+        int r;
+
+        assert(address);
+        assert(link);
+
+        addr = address->in_addr.in6;
+        ORDERED_HASHMAP_FOREACH(j, link->network->ipv6_tokens, i) {
+                if (j->address_generation_type == IPV6_TOKEN_ADDRESS_GENERATION_PREFIXSTABLE
+                    && memcmp(&j->prefix, &addr, FAMILY_ADDRESS_SIZE(address->family)) == 0) {
+                        for (; j->dad_counter < DAD_CONFLICTS_IDGEN_RETRIES_RFC7217; j->dad_counter++) {
+                                r = ipv6_makestableprivate(link, &addr, prefixlen, &j->dad_counter, &address->in_addr.in6);
+                                if (r < 0)
+                                        return r;
+
+                                if (!ipv6_interface_identifiers(&address->in_addr.in6)) {
+                                        prefix_stable = true;
+                                        break;
+                                }
+                        }
+                } else if (j->address_generation_type == IPV6_TOKEN_ADDRESS_GENERATION_EUI64) {
+                        memcpy(((char *)&address->in_addr.in6) + 8, ((char *)&j->prefix) + 8, 8);
+                        break;
+                }
+        }
+
+        /* eui64 or fallback if prefixstable do not match */
+        if (!prefix_stable) {
```

Comment:
Hmm. If the type is eui64, then the token seems always ignored. I guess the flag `prefix_stable` (I think it should be renamed though) should be also set for when eui64.

Reply:
how ?

Reply:
When EUI64, once token is written to `address`, but after that the token is overwritten based on the MAC address, as `prefix_stable` is not set in that case. Do I misunderstand something?